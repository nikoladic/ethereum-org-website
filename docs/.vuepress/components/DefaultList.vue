<template>
<div>
	<ul>
		<li v-for="item in this.items">
			<a :href="item.url" target="_blank" rel="noopener noreferrer">{{item.name}}</a>, {{item.description}} 
		</li>
	</ul>
</div>
</template>

<script>

export default {
	name: 'default-list',
        props: {
                items: Array
        }	
}

</script>

