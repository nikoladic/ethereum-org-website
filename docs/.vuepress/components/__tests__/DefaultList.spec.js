import { shallowMount } from '@vue/test-utils'
import DefaultList from "@/components/DefaultList"

describe('DefaultList', () => {
  test('list items', () => {
    const wrapper = shallowMount(DefaultList, {
      propsData: {
        items: [{
          name: "Bitcoin",
          description: "The 1st Crytocurrency in the world"
        },{
          name: "Ethereum",
          description: "The smart contract blockchain"
        }]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('list no items', () => {
    const wrapper = shallowMount(DefaultList, {
      propsData: {
        items: []
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
