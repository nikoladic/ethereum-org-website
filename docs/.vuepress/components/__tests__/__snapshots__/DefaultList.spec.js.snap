// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultList list items 1`] = `
<div>
  <ul>
    <li>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        Bitcoin
      </a>
      , The 1st Crytocurrency in the world
    
    </li>
    <li>
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        Ethereum
      </a>
      , The smart contract blockchain
    
    </li>
  </ul>
</div>
`;

exports[`DefaultList list no items 1`] = `
<div>
  <ul />
</div>
`;
